prov/mrail/src/mrail_init.c

```diff
@@ -69,15 +69,30 @@ static int mrail_parse_env_vars(void)
 	return 0;
 }
 
-int mrail_get_core_info(uint32_t version, const char *node, const char *service,
-			uint64_t flags, const struct fi_info *hints,
-			struct fi_info **core_info)
+static int mrail_count_info(struct fi_info *info)
 {
-	struct fi_info *core_hints, *info, *fi = NULL;
-	size_t i;
+	int n = 0;
+
+	while (info) {
+		n++;
+		info = info->next;
+	}
+
+	return n;
+}
+
+static int mrail_get_core_info(uint32_t version, const char *node, const char *service,
+			       uint64_t flags, const struct fi_info *hints,
+			       struct fi_info ***core_info_array, int *num_core_info)
+{
```

Comment:
This function is getting long.  Can you move some of the functionality into sub-calls?

Reply:
Sure, will break it up.